### DISCUSSION TITLE
Issue with Upgrade

### DISCUSSION THREAD

**Discussion by StodgyWaif** [thorrak/fermentrack#776] [Q&A]
Hello.
I am running Fermentrack in docker on Ubuntu 20.04.6 in Hyper-v.
It has been running well for a long time but recently I started getting an nginx 502 error on the web portal.
I decided to try and do an upgrade because I was overdue for one anyway.

I started the install command `curl -L install.fermentrack.com | bash` but it seems to be stuck at:
`Waiting for PostgreSQL to become available...`

I waited for a long time and restarted the server and the install script but it just goes back to this message.

Anything else I could try?

**Comment by vanosg** [comment]
Hold old is the last install? A lot has changed... just throwing spaghetti at a wall until someone else chimes in.  A log would help better identify where in the process you are and where the issue is. I know when I got a little sideways in an upgrade process the  there were two instances of postgreql trying to run (I can't seem to find my issue discussing it here for some reason though...), if i recall correctly one was started by the circus supervisor process, and one started by docker/fermentrack/something else.

Post a log when you get a chance! ``docker ps`` will show if all images are running properly.  Or maybe remove all your docker images (``docker rmi``) and run the install process again. If this instance had a pre-docker install running, that too might still be causing issues as the old circus supervisor process may be trying to start its own process on the host, instead of docker, and hogging the port docker is trying to use, hence stopping the docker instance from starting. Lots of ideas, logs can help :)  Or just flash the SD card and start from scratch if that's an option for you as well- not ideal, but may just be quicker.

**Reply by StodgyWaif** [reply]
Hi. Thanks for replying.
I think it hasn't been updated in 6 months or so but I can't be sure about that.
I did take a VM checkpoint before I started the upgrade so I can revert it although that will just take me back to the 502 error state I started in.

So I've used Ubuntu quite a bit but I'm certainly not an expert. And I definitely have less experience still with docker.
Say I remove the docker images, will the data/configuration be retained? Or is there a way to extract the configuration and data and import them into a new instance? I'm guessing I need to back up the postgresql database but is there anything else?

Honestly if I lost everything it's not a huge deal. I guess I'm just looking for the easiest way to resolve this.

**Comment by vanosg** [comment]
Again I'm just a guy on the internet, but looking at the docker-compose file it looks like @thorrak set it up to mount external volumes (not stored inside the image/container) to handle data, logs, and backup files. It wouldn't hurt to make your own copies of this data if you can (perhaps look at ``docker cp`` command as one way to get the files out of the running container and onto the host), but I would feel confident to say that removing the images and reinstalling using the curl command SHOULD preserve your data... but I'd still make a backup just in case!

I'll let someone else answer if the right path to upgrade is to run the install script again, or if there's a better way- it used to be via the Web UI, but not sure of the intended method now that the switch to docker occurred. Good luck

**Comment by StodgyWaif** [comment]
Thank you for the suggestions.
After learning a few basic things about docker I decided to roll back my changes to when I had the 502 error.
Turns out the postgres and django containers were not running.
Once I started those I could see the web portal again.

Now I'll need to try and tackle the upgrade but at least I've got it back to a working state again.

Thanks again

**Comment by vanosg** [comment]
On the command line:
```
cd ~/fermentrack-tools
docker-compose pull
docker-compose up -d
```

should do the trick to update the images
